snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -223,10 +223,11 @@ def _wait_for_network(self):
         logger.info('Network connection established')
 
     def _inject_snapcraft(self):
-        if common.is_snap():
-            # Because of https://bugs.launchpad.net/snappy/+bug/1628289
-            self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
+        # Because of https://bugs.launchpad.net/snappy/+bug/1628289
+        # Needed to run snapcraft as a snap and build-snaps
+        self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
```

Comment:
Now this is not only related to snapcraft. Shouldn't we move it to another function? `inject_snapcraft` doesn't sound like the right place for this to be.

Reply:
Agreed.

Reply:
@kyrofa @elopio if you guys are going to make a comment that requires a change mark the PR appropriately.

Reply:
I think we both just wanted to hear justification for putting it here, not necessarily requiring a change.

Reply:
I moved it to the respective container creation code paths of the two classes now.